### ISSUE TITLE
Unsafe.ReadUnaligned not working on armv5

### ISSUE THREAD

**Issue by topschrott** [mono/mono#19142]
I'm getting wrong results when using the `BitConverter` to extract a 16-bit or 32-bit integer on my ARMv5 platform since moving from Mono 5 to version 6.

In Mono version 6, the `BitConverter` is using `Unsafe.ReadUnaligned` instead of implementing it directly, as in version 5. 

To me, it looks like `Unsafe.ReadUnaligned` is not working properly and a simple type-cast is used from an unaligned address. Of course it is also possible that I am missing a build-option when compiling Mono?

Issue #18221 might be related, but I am not using the llvm.

## Steps to Reproduce

I'm using the following test code:

```c#
byte[] buf = new byte[] {0x11, 0x22, 0x33, 0x44, 0x55};

for (int i=0; i<4; i++) {
    int v = BitConverter.ToUInt16(buf, i);
    Console.WriteLine(String.Format("{0}: {1:X}", i, v));
}
```

The output on the ARM platform is:

```
0: 2211
1: 2211
2: 4433
3: 4433
```

Of course, I expect it to say:

```
0: 2211
1: 3322
2: 4433
3: 5544
```

The Mono runtime is built by Buildroot with the following configure options:

```
        --with-mcs-docs=no \
        --enable-small-config=yes \
        --with-ikvm-native=no \
        --enable-minimal=attach,com,debug,profiler,shadow_copy,simd \
        --enable-static
```

I'm executing my test with:

```sh
mono -O=-float32 test.exe
```

## On which platforms did you notice this

[ ] macOS
[x] Linux
[ ] Windows

**Version Used**:

```sh
mono --version
Mono JIT compiler version 6.8.0.105 (tarball Fri  6 Mar 11:04:59 UTC 2020)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       normal
        Notifications: epoll
        Architecture:  armel,vfp+fallback
        Disabled:      attach,com,debug,profiler,shadow_copy,simd
        Misc:          smallconfig softdebug 
        Interpreter:   yes
        LLVM:          supported, not enabled.
        Suspend:       preemptive
        GC:            sgen (concurrent by default)
```

**Comment by EgorBo** [comment]
Most likely fixed in https://github.com/mono/mono/pull/18296 cc @imhameed

**Comment by topschrott** [comment]
I believe that the version I am using (6.8.0.105) has the fix from #18296 included. But I'm not using the LLVM, so not sure if it would apply anyway?

**Comment by imhameed** [comment]
#18296 is LLVM-only and doesn't affect this bug.

I'm taking a quick glance at all of this and it looks like our native backend doesn't do anything special for `MONO_INST_UNALIGNED` load mini-ir ops. `Unsafe.ReadUnaligned` should be treated as an intrinsic regardless of codegen backend: https://github.com/mono/mono/blob/mono-6.8.0.105/mono/mini/intrinsics.c#L528. But after this point nothing else checks that flag.

The handling of this flag seems to be done in memory-access.c, as part of `mini_emit_memory_load` and `mini_emit_memory_store`, but these are called by `method_to_ir`, and I think those codepaths would only fire for the `unaligned` CIL prefix instruction. It'd be nice if these things (and also volatile-prefix CIL instructions vs intrinsic atomic loads and stores) were handled in a more unified way.
